pytorch_lightning/trainer/data_loading.py

```diff
@@ -139,17 +143,26 @@ def auto_add_sampler(
                     ' distributed training. Either remove the sampler from your DataLoader or set'
                     ' `replace_sampler_ddp`=False if you want to use your custom sampler.'
                 )
-
-            # replace with distributed sampler
             sampler = self._get_distributed_sampler(dataloader, shuffle, mode=mode)
-            dataloader = self.replace_sampler(dataloader, sampler, mode=mode)
+        else:
+            # use current sampler
+            sampler = dataloader.sampler
+
+        dataloader = self.replace_sampler(dataloader, sampler, mode=mode)
 
         return dataloader
 
     @staticmethod
-    def _resolve_batch_sampler(dl_args, dataloader, sampler, mode: Optional[RunningStage] = None) -> Dict[str, Any]:
+    def _resolve_batch_sampler(
+        dl_args,
+        dataloader,
+        sampler,
+        mode: Optional[RunningStage] = None,
+        should_use_forward_sampler: Optional[bool] = False
```

Comment:
Is this an parameter we will want when fault tolerance is enabled by default? If not, it should be removed